src/schema.rs

```diff
@@ -453,6 +486,51 @@ impl Schema {
     /// Avro supports "recursive" definition of types.
     /// e.g: {"type": {"type": "string"}}
     fn parse_complex(complex: &Map<String, Value>) -> Result<Self, Error> {
+        fn logical_verify_type(complex: &Map<String, Value>, t: &str) -> Result<(), Error> {
+            match complex.get("type") {
+                Some(&Value::String(ref a)) if a == t => Ok(()),
+                Some(&Value::String(ref a)) => Err(ParseSchemaError::new(format!(
+                    "Unexpected `type` ({}) variant for logicalType",
+                    a
+                )))?,
+                _ => Err(ParseSchemaError::new(
+                    "Unexpected type variant for logicalType",
+                ))?,
+            }
+        }
+        match complex.get("logicalType") {
+            Some(&Value::String(ref t)) => match t.as_str() {
+                "decimal" => unimplemented!("TODO - XXX `decimal` logicalType"),
```

Comment:
I guess this is one of unimplemented features, the `decimal` logical type. We could implement it in a later PR, I suppose.

Reply:
I'm happy to implement it here.

Reply:
Great!